opslevel/resource_opslevel_group.go

```diff
@@ -0,0 +1,189 @@
+package opslevel
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/opslevel/opslevel-go"
+	"github.com/shurcooL/graphql"
+)
+
+func resourceGroup() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a group",
+		Create:      wrap(resourceGroupCreate),
+		Read:        wrap(resourceGroupRead),
+		Update:      wrap(resourceGroupUpdate),
+		Delete:      wrap(resourceGroupDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"last_updated": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"alias": {
+				Type:        schema.TypeString,
+				Description: "The human-friendly, unique identifier for the group.",
+				Computed:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The name of the group.",
+				ForceNew:    false,
+				Required:    true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "The description of the group.",
+				ForceNew:    false,
+				Optional:    true,
+			},
+			"parent": {
+				Type:        schema.TypeString,
+				Description: "The parent of the group.",
+				ForceNew:    false,
+				Optional:    true,
+			},
+			"members": {
+				Type:        schema.TypeList,
+				Description: "The users who are members of the group.",
+				ForceNew:    false,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"teams": {
+				Type:        schema.TypeList,
+				Description: "The teams where this group is the direct parent.",
+				ForceNew:    false,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceGroupCreate(d *schema.ResourceData, client *opslevel.Client) error {
+	members := []opslevel.MemberInput{}
+	for _, member := range getStringArray(d, "members") {
+		members = append(members, opslevel.MemberInput{Email: member})
+	}
+
+	teams := []opslevel.IdentifierInput{}
+	for _, team := range getStringArray(d, "teams") {
+		if opslevel.IsID(team) {
```

Comment:
Lets convert this to be

```
teams = append(teams, opslevel.NewIdentifier(team))
```

Because that takes care of the If IsID checking automatically

Reply:
Resolving since this should be done